plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/wizard/NewServerWizard.java

```diff
@@ -35,15 +37,28 @@ public NewServerWizard() {
 	@Override
 	public void finish(TimePeriod timeout) {
 		// workaround due to JBDS-3596
-		new WaitUntil(new JobIsKilled("Refreshing server adapter list"), TimePeriod.LONG, false);
+		handleRefereshingServersJob();
 		super.finish(timeout);
 	}
 
 	@Override
 	public void cancel() {
 		// workaround due to JBDS-3596
-		new WaitUntil(new JobIsKilled("Refreshing server adapter list"), TimePeriod.LONG, false);
+		handleRefereshingServersJob();
 		super.cancel();
 	}
 	
+	private void handleRefereshingServersJob(){
```

Comment:
Please avoid duplicates of the same logic. Consider creating a new WaitCondition, e.g. ServerAdaptersAreRefreshing.